src/System Application/Test Library/User Details/src/UserDetailsTestLibrary.Codeunit.al

```diff
@@ -0,0 +1,113 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.TestLibraries.Security.User;
+
+using System.Security.User;
+
+/// <summary>
+/// Test library for the User Details module
+/// </summary>
+codeunit 132001 "User Details Test Library"
+{
+    var
+        UserDetailsRec: Record "User Details";
+        UserDoesNotExistErr: Label 'The user with security ID %1 does not exist', Locked = true;
+
+    /// <summary>
+    /// Saves the user details inside this instance, so that they can be access later.
+    /// </summary>
+    procedure FetchUserDetails()
+    var
+        UserDetails: Codeunit "User Details";
+    begin
+        UserDetails.GetUserDetails(UserDetailsRec);
+    end;
+
+    /// <summary>
+    /// Gets the value of "Has SUPER permission set" for the user.
+    /// </summary>
+    /// <param name="UserSID">The user security ID</param>
+    procedure GetHasSuperPermissionSet(UserSID: Guid): Boolean
+    begin
+        if not UserDetailsRec.Get(UserSID) then
+            Error(UserDoesNotExistErr, UserSID);
+
+        UserDetailsRec.CalcFields("Has SUPER permission set");
```

Comment:
If you instead set auto-calcfield on this field before doing a GET, you automatically get this value evaluated without having to do a second database call here. I know this isn't super important for tests, just wanted to mention this speedup :)

Reply:
I actually tried that originally, but it didn't work for some reason. As this is very monor - resolving.